UnityProject/Assets/Scripts/Communications/SignalEmitter.cs

```diff
@@ -4,28 +4,53 @@
 using ScriptableObjects.Communications;
 using UnityEngine;
 using Mirror;
+using NaughtyAttributes;
 
 namespace Communications
 {
 	public abstract class SignalEmitter : NetworkBehaviour
 	{
-		[SerializeField] protected SignalDataSO signalData;
-		[SerializeField] protected float frequency = 122f;
+		[SerializeField]
+		[Required("A signalSO is required for this to work.")]
+		protected SignalDataSO signalData;
+		public EncryptionDataSO EncryptionData;
```

Comment:
if not because you like encapsulation, just because it is more explicit prefer to make a private field with SerializedField attribute and expose the get/set as a property.